packages/lib/SyncTargetOneDrive.ts

```diff
@@ -103,7 +103,11 @@ export default class SyncTargetOneDrive extends BaseSyncTarget {
 		}
 		this.api_.setAccountProperties(accountProperties);
 		const appDir = await this.api().appDirectory();
-		const fileApi = new FileApi(appDir, new FileApiDriverOneDrive(this.api()));
+		// the appDir might contain non-ASCII characters
+		// /[^\u0021-\u00ff]/ is used in Node.js to detect the unescaped characters.
+		// See https://github.com/nodejs/node/blob/bbbf97b6dae63697371082475dc8651a6a220336/lib/_http_client.js#L176
+		const baseDir = RegExp(/[^\u0021-\u00ff]/).exec(appDir) !== null ? encodeURI(appDir) : appDir;
+		const fileApi = new FileApi(baseDir, new FileApiDriverOneDrive(this.api()));
```

Comment:
It's strange that we'd need to encode it in some cases and not others. Shouldn't it be always encoded?

Reply:
Sometimes we will get the encoded result, but sometimes we won't.
For example, in #7851, They both use the work/education account, which belong to specific organizations, and they get the raw URL. However, I use a personal account which doesn't belong to any organization, and I get the encoded URL.
The inconsistency does exist so I need to detect if the `appDir` is encoded or not.

Reply:
Do you mean that the OneDrive API sometimes return an encoded URL and sometimes it doesn't? Or are we reading the URL from the wrong place in some cases?

Reply:
> Do you mean that the OneDrive API sometimes return an encoded URL and sometimes it doesn't?

I mean this one. Sorry for my bad English.